operators/ehr_query_llm/fhir/resource_sanitizer.py

```diff
@@ -0,0 +1,658 @@
+                             second value is the actual data.
+                             When content type is text, the data is the actual decoded value.
+                             Otherwise, the value is in the original form which could be base64 encoded string.
+        """
+        if attachment.contentType == "text/plain; charset=utf-8":
+            return (attachment.contentType, base64.b64decode(attachment.data).decode("utf-8"))
+        else:
+            return (attachment.contentType, attachment.data)
+            # raise NotImplementedError(f"Content type not supported: {attachment.contentType}")
+
+    @staticmethod
+    def _get_coding_value(coding: Coding) -> str:
+        """Returns the display value of a given Coding object when available. Otherwise, return the actual code."""
+        return coding.display if coding.display else coding.code
+
+    @staticmethod
+    def _get_name(name: HumanName) -> str:
+        """Returns a formatted name of a given HumanName"""
+
+        def _prefix(prefixes) -> str:
+            return ",".join(prefixes) if prefixes else ""
+
+        return f"{_prefix(name.prefix)} {name.family}, {' '.join(name.given)}".strip()
+
+    @staticmethod
+    def _get_marital_status(marital_status: str) -> str:
+        """Converts abbreviated martial status to actual meaning."""
+        if marital_status and marital_status.text:
+            mapping = {
+                "A": "Annulled",
+                "D": "Divorced",
+                "I": "Interlocutory",
+                "L": "Legally Separated",
+                "M": "Married",
+                "C": "Common Law",
+                "P": "Polygamous",
+                "T": "Domestic partner",
+                "U": "Unmarried",
+                "S": "Never Married",
+                "W": "Widowed",
+            }
+
+            text = marital_status.text.upper()
+            return mapping.get(text, text)
+        return "Unknown"
+
+    @staticmethod
+    def _new_resource(original_data: Dict, model: FHIRAbstractModel) -> Dict:
+        """Creates a sanitized resource dictionary with common/shared attributes"""
+        sanitized_data = {}
+        resource = {}
+        sanitized_data["resource"] = resource
+        sanitized_data["fullUrl"] = original_data.get("fullUrl", "N/A")
+        resource["resourceType"] = model.resource_type
+        resource["id"] = model.id
+        return sanitized_data, resource
+
+    @staticmethod
+    def _allergy_reaction(reactions: AllergyIntoleranceReaction) -> Dict:
+        """Parses AllergyIntoleranceReaction"""
+        data = {}
+        for reaction in reactions:
+            texts = [item.text for item in reaction.manifestation]
+            data[",".join(texts)] = reaction.severity
+        return data
+
+    @staticmethod
+    def _remove_html_tags(text) -> str:
+        """Removes HTML tags"""
+        tags = re.compile("<.*?>")
+        return re.sub(tags, "", text)
+
+
+if __name__ == "__main__":
```

Comment:
Could we add a unit test to run and validate this file?

Reply:
It is definitely required to have a "unit" test, and for this module the main() function is served as the test function. This module's main functionality to parse the EHR records defined by FHIR Standards R4, for a defined list of record types. Given that the specific version of the Standards defined schema will never change, code logic for validating mandatory and optional attributes as well as mapping attributes will largely remain the same. An instance of each supported record types is added to the example JSON file to use as test data if the logic ever changes, with the file size totaling 15KB. Instruction is given in the code for downloading large public dataset. So, given the nature of this module, the current testing setup with main() is deemed sufficient.

Reply:
Got it. It would be good to revisit unit tests eventually to understand how we can cover this work in automation, I'm not clear at the moment on how to validate the individual components. But, if your and @cdinea 's integration efforts have proven successful then the automation concern doesn't need to block this initial contribution.